### ISSUE TITLE
[Game Support]: Limbus Company

### ISSUE THREAD

**Issue by the-jason-json** [PlayCover/PlayCover#1585]
### App Store link

https://apps.apple.com/us/app/limbus-company/id6444112366

### Current issues

- [ ] Game fails to install
- [ ] Game fails to open
- [X] Game crashes randomly during gameplay
- [ ] Game does not work with keymapping
- [ ] Game experiences graphical issues or broken UI
- [ ] Game fails to connect to servers or complete download

### More details

Limbus Company crashes after opening a second time (Have to clear playchain to open again without crashes)

Removing playtools breaks the login, as clicking the apple login does nothing(it also does nothing when playchain's injected, but before updating to 3.0 apple login worked), and the google login opens the safari login, but after logging in the game stays in an un-logged in state(google login works when playchain is injected)

The attached video is just to show that Limbus Company crashes when some bypasses aren't turned on, even when the options are greyed out.
If further videos of the issue are needed, please tell me.
Crashlog is too long to send in the crashlog section, hence the .txt

[Limbus Crashlog.txt](https://github.com/user-attachments/files/16186032/Limbus.Crashlog.txt)
https://github.com/user-attachments/assets/f8a83fd9-b4c6-4a00-8e99-e69d384b8bba

### What version of PlayCover are you using?

3.0.0

### What version of macOS are you using?

Sonoma (macOS 14)

### Issue Language

- [X] Yes my issue is written in English

**Comment by Depal1** [comment]
> Removing playtools breaks the login, as clicking the apple login does nothing(it also does nothing when playchain's injected, but before updating to 3.0 apple login worked), and the google login opens the safari login, but after logging in the game stays in an un-logged in state(google login works when playchain is injected)

This behaviour is fully expected with some caveats. Removing PlayTools may break the login functionality because you are relying on PlayChain to store your credentials, and PlayChain is an integral part of PlayTools. 

The same goes for the Google login.

Apple’s login will only work with SIP and AMFI disabled since it requires entitlements only available with a certificate from a paid Apple Developer account.

**Comment by Depal1** [comment]
Why do you want to run it with PlayTools disabled?

**Comment by Depal1** [comment]
Taking into account what was just stated, what is worth checking out is
> Limbus Company crashes after opening a second time (Have to clear playchain to open again without crashes)

Can this be consistently replicated when PlayTools isn’t being removed?

**Comment by the-jason-json** [comment]
Disabling AMFI seems to break Zenless Zone Zero though, so not an option ;-;
Removing Playtools fixed Limbus in a previous nightly version though iirc (Also inject playtools requires me to command+r Limbus a few times before it's in the right orientation, just mentioning cause maybe there's a fix for this that I missed)
Yes, when Playtools is injected, it still crashes when opening unless I clear playchain, and then it only opens stably once, before having to clear playchain the next time.

**Comment by the-jason-json** [comment]
Correction to previous statement, it only crashes again when after I login. I can consistently open Limbus Company without crashing if I don't login.

**Comment by the-jason-json** [comment]
After some testing (credits to ohaiibuzzle), it seems like the problem's with the Firebase SDK crashing.
![image](https://github.com/user-attachments/assets/116fd8bc-b85c-4ac3-b6fa-a75acb48f038)

After wiping the 2 firebase keys, then opening the game, logging in and closing it, the firebase keys are recreated.
Oh yeah might as well mention, the game launched sideways. It can be solved with command+R a few times, but just mentioning it in case it's a bug that can be fixed
![image](https://github.com/user-attachments/assets/4891b742-b9ed-4e0f-b0c8-606fbd40f3f7)

**Comment by PenetraP** [comment]
same problem happened.
may i know how to wipe the firebase keys?

**Comment by FahrezyAli** [comment]
> After some testing (credits to ohaiibuzzle), it seems like the problem's with the Firebase SDK crashing. ![image](https://github.com/user-attachments/assets/116fd8bc-b85c-4ac3-b6fa-a75acb48f038)
> 
> After wiping the 2 firebase keys, then opening the game, logging in and closing it, the firebase keys are recreated. Oh yeah might as well mention, the game launched sideways. It can be solved with command+R a few times, but just mentioning it in case it's a bug that can be fixed ![image](https://github.com/user-attachments/assets/4891b742-b9ed-4e0f-b0c8-606fbd40f3f7)

Question, how to do this? And by doing this, does it fix the crashing on the second login issue?

**Comment by the-jason-json** [comment]
> 
> Question, how to do this? And by doing this, does it fix the crashing on the second login issue?

If you just need to fix the crashing issue, just clear playchain data before opening limbus and it should work

<img width="230" alt="Image" src="https://github.com/user-attachments/assets/b08a7361-7cd0-4142-9cbe-7e804f692626" />

**Comment by FahrezyAli** [comment]
> > Question, how to do this? And by doing this, does it fix the crashing on the second login issue?
> 
> If you just need to fix the crashing issue, just clear playchain data before opening limbus and it should work
> <img alt="Image" width="230" src="https://private-user-images.githubusercontent.com/106799225/419757337-b08a7361-7cd0-4142-9cbe-7e804f692626.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEyNDA3OTksIm5iZiI6MTc0MTI0MDQ5OSwicGF0aCI6Ii8xMDY3OTkyMjUvNDE5NzU3MzM3LWIwOGE3MzYxLTdjZDAtNDE0Mi05Y2JlLTdlODA0ZjY5MjYyNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwNlQwNTU0NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMmUwMjg1YzA5MzVjOGM4MWRkZGI1MjNiNzA4YzUxYmFlYzgxMWVjMWIzNDhkNTBkZjg4YzU5ODUxNDA4NTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.x5N47keqc8BHsYCXXBvI-82MlDHb_hGftPNGfVDlWq4">

Does this mean you need to log in every time you launch the game?

**Comment by the-jason-json** [comment]
> Does this mean you need to log in every time you launch the game?

Yes. Someone on Discord mentioned they didn't have to, so ig you could fiddle with some settings to fix it but idk
